kaggle_environments/agent.py

```diff
@@ -112,20 +112,32 @@ def act(self, observation, timeout=10):
         if self.agent is None:
             self.agent = build_agent(self.raw, self.environment)
             # Add in the initialization timeout since this is the first time this agent is called
-            timeout += self.configuration.agentTimeout
+            timeout += self.environment.configuration.agentTimeout
 
         args = [
            structify(observation),
-           structify(self.configuration)
+           structify(self.environment.configuration)
         ][:self.agent.__code__.co_argcount]
 
-        try:
-            action = self.agent(*args)
-        except Exception as e:
-            action = e
+        with StringIO() as out_buffer, StringIO() as err_buffer, redirect_stdout(out_buffer), redirect_stderr(err_buffer):
+            try:
+                action = self.agent(*args)
+            except Exception as e:
+                action = e
+            # Allow up to 1k log characters per step which is ~1MB per 600 step episode
+            max_log_length = 1024
+            out = out_buffer.getvalue()[0:max_log_length]
+            err = err_buffer.getvalue()[0:max_log_length]
+
+        duration = time() - start
+        log = {
```

Comment:
This will be the json format of each log step, and the overall log file will be a list of lists of these logs where the outer list has a list for each step and the inner list has a log for each player for that step.

Reply:
`[[{"duration":1.123,"stdout":"Hello world!\n","stderr":""}, {...}, {...}, {...}], [...], ...]`